src/main/java/org/galatea/starter/domain/RangeType.java

```diff
@@ -0,0 +1,46 @@
+package org.galatea.starter.domain;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+public enum RangeType {
```

Comment:
This looks good! You can take it one step further though so that you can just specify a lambda rather than having to write the full method each time.

Make the subtract() method final and change its type to Function<Integer, LocalDate>. It doesn't need to be abstract any more either. Add it into the constructor.

Then, in the enum constructors, you can do:
Days("d", rangeValue -> LocalDate.now().minusDays(rangeValue)).

Thus setting the subtract() method to the lambda you pass in.

Reply:
Done.